### ISSUE TITLE
508-defect-2 [COGNITION] Find a VA Form sort feature - Add code to announce list re-sort to screen reader users

### ISSUE THREAD

**Issue by MarciMcGuireGCIO** [department-of-veterans-affairs/va.gov-team#19157]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _First name only. Delete if not applicable._

## Details

Currently, post-selection of a sort, screen reader users do not get any clear feedback that the search results have been updated. An alert or announcement post-selection would help make up for the absence of an explicit submit button.

@zacharymorel has a fix to address this

## Acceptance Criteria

- [ ] Screen reader users receive feedback post-sort selection that their search results have been updated e.g. "Showing x results sorted by newest..."

**Comment by MarciMcGuireGCIO** [comment]
New issue related to this ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/19117#issuecomment-768462484

**Comment by MarciMcGuireGCIO** [comment]
Thank you @joshkimux !!

**Comment by zacharymorel** [comment]
FYI, I have to check with VSP engineering. I have been mistaken and this looks like it will fall under their wheel house.

**Comment by zacharymorel** [comment]
@joshkimux is there another way to solve this incase we cannot add this to the VSP maintained component-library?

**Comment by joshkimux** [comment]
@zacharymorel We could set focus to the results string using `tab-index="-1"`? I don't have a confident answer to this yet. Let me check in with @1Copenut to see how he might approach this.

**Comment by zacharymorel** [comment]
Oh okay, I can just this a try then.

**Comment by zacharymorel** [comment]
Okay, looks like I will be working with the VSP engineers on the design library team to make some adjustments to the Select component. I will be able to use the original solve (using aria-content and aria-live attributes).
@joshkimux is that something we need Trevors blessing for as well?

**Comment by joshkimux** [comment]
No, this should be ok! @1Copenut tagging you just for awareness 🥳

**Comment by zacharymorel** [comment]
@joshkimux As far as using the aria-live attribute, should we prefer **assertive** or **polite**?

**Comment by joshkimux** [comment]
@zacharymorel Let's use `assertive` 👍

**Comment by 1Copenut** [comment]
@joshkimux && @zacharymorel I'll add myself as a reviewer to the PR. I have a couple of questions that will be easier to track there. I think the proposed solution will work fine, as long as the live region is on the page on first render. Screen readers (from what I can gather) do a pass over the page and make note of live regions and the initial state, so they have something to react to when the live regions change.

Let's plan to test the `assertive` setting when the UI starts coming together. It feels like the right approach; trying it out with AT will confirm it.

Thank you both!

**Comment by joshkimux** [comment]
@zacharymorel @MarciMcGuireGCIO @1Copenut 

Aside from this ticket, I don't have anything else to add on to the staging review for the VA form sort feature. However, some new considerations emerged from testing using JAWS/Chrome, JAWS/FF, NVDA/Chrome, and NVDA/FF on a Windows HP laptop.

Unlike Chrome/Safari on Mac, Chrome/FF on Windows lets users to move through options and make selections using arrow keys. This is different than Chrome/Safari on Mac which forces users to make an explicit `enter` or `space` selection on the option they prefer. (see screenshots below)

That being said, a couple of new considerations come to mind:
- Having an announcement on selection be made to screen reader users is even more critical for instances where selections are made using the arrow keys as it is not as clear that updates will be made on the page without an explicit `enter` or `space`
- We may want to consider re-arranging the string format of Last Updated (Newest) to the opposite: Newest (Last Updated). Since "Newest" and "Oldest" are the delta between the two options, not last updated, announcing them first increases the likelihood screen reader users will hear the type of selection. Furthermore, the important part of the string will be easier to understand for JAWS users as JAWS announces "()" as "left parent" and "right parent." "Last updated left parent newest right parent" is may be less effective at conveying meaning than "Newest left parent last updated right parent"

---

### Explicit choice required on Safari/VO on Mac
<img width="676" alt="Screen Shot 2021-02-03 at 1 08 58 PM" src="https://user-images.githubusercontent.com/14154792/106790065-0b089d80-6621-11eb-8f82-d815c5882057.png">

---

### Arrow keys can change selections on Jaws/NVDA on Windows**

https://user-images.githubusercontent.com/14154792/106791342-cc73e280-6622-11eb-870a-0598824ae8e9.mov

**Comment by zacharymorel** [comment]
FYI, the Pull Request once merged, auto closed this. If we want to continue discussion, feel free to reopen!
